extra_geom/base.py

```diff
@@ -659,7 +659,10 @@ def to_pyfai_detector(self):
             raise NotImplementedError
 
         from . import pyfai
-        det = getattr(pyfai, self._pyfai_cls_name)()
+        if 'JUNGFRAU' in self.detector_type_name:
+            det = getattr(pyfai, self._pyfai_cls_name)(n_modules=self.n_modules)
```

Comment:
In general, the way to have special behaviour for one subclass like this is to override the method in that subclass, rather than putting an `if` in the base class. The idea is that the base class doesn't 'know' about its subclasses.

Reply:
Thank you, implemented in https://github.com/European-XFEL/EXtra-geom/pull/197/commits/7a3d83e882ced59524f527dca74dac1249b8ed84.